### ISSUE TITLE
output_notebook hangs on loading BokehJS and says javascript is disabled

### ISSUE THREAD

**Issue by jwhendy** [bokeh/bokeh#7173]
The [docs](https://bokeh.pydata.org/en/latest/docs/user_guide/notebook.html) suggest that output in `jupyterlab` should be possible via `output_notebook()`. Trying that way:

```
from bokeh.io import output_notebook
output_notebook()

Loading BokehJS ...
JavaScript output is disabled in JupyterLab
```

There's also [this comment](https://github.com/bokeh/bokeh/issues/6700#issuecomment-338214632) which has the same results as me using this (note importing `output_notebook` from  `bokeh.plotting` vs. `bokeh.io`):

```
from bokeh.plotting import output_notebook
output_notebook()

### same error as above
```

It would appear that the [only problem](https://github.com/bokeh/bokeh/issues/6700#issuecomment-340726381) with the last example was the user being on an old version. I don't *think* that's going on in my case. My info:

From within `jupyter lab`:
```
import bokeh
print(bokeh.__version__)
0.12.10
```

Various relevant package versions:
```
$ pip freeze
bokeh==0.12.10
ipython==6.2.1
jupyter-client==5.1.0
jupyter-console==5.2.0
jupyter-core==4.4.0
jupyterlab==0.28.12
jupyterlab-launcher==0.5.5

$ jupyter labextension install --app-dir=~/.local/share/jupyter/lab jupyterlab_bokeh
$ jupyter labextension list --app-dir=~/.local/share/jupyter/lab
JupyterLab v0.28.12
Known labextensions:
   app dir: /home/jwhendy/.local/share/jupyter/lab
jupyterlab_bokeh
        jupyterlab_bokeh v0.1.0  enabled  OK

### I'm starting jupyterlab like this:
$ jupyter lab --dev-mode --app-dir=~/.local/share/jupyter/lab
```

Should this work, or is there something else I'm not checking/setting up properly?

**Comment by bryevdv** [comment]
Does it work without "--dev-mode" I have no idea what that is or does. Bokeh sends both the old JS payload (for classic notebook) and a new mimetype (for jlab) so if it is saying "JavaScript output is disabled in JupyterLab" then that means it does not recognize or see the bokeh jlab extension.

**Comment by bryevdv** [comment]
FWIW currently working for me as expected:

<img width="899" alt="screen shot 2017-11-05 at 15 16 15" src="https://user-images.githubusercontent.com/1078448/32419219-5014ff68-c23c-11e7-8ffa-709be0d88878.png">

**Comment by jwhendy** [comment]
What version of `jupyterlab` are you on? I updated recently and without `--dev-mode`, I get:
```
500 : JupyterLab Error
The error was:
JupyterLab assets not detected in "/usr/share/jupyter/lab/static"
```

When I googled that yesterday, I found various suggestions for either `--dev-mode` or `--core-mode`. 

Grrr. So in re-reading [this comment](https://github.com/jupyterlab/jupyterlab/issues/3136#issuecomment-338791916) for maybe the 10th time, I spotted `jupyter lab build` as a second option. All my stuff is installed locally (`pip install --user`), and I didn't want to install to system in order to prevent package manager clobbering so I've been using `app-dir=~/.local/share/jupyter/lab` for any of my `jupyter foo` commands.

I just did `jupyter build --app-dir=~/.local/share/jupyter/lab`, which worked, and now `jupyter lab --app-dir=~/.local/share/jupyter/lab` 1) starts successfully without `--dev-mode` **and** 2) things render now.

Hopefully this helps others and sorry for the noise!
